tutorial/10_key_features/005_visualization.py

```diff
@@ -0,0 +1,163 @@
+
+        optimizer.zero_grad()
+        output = model(data)
+        loss = F.nll_loss(output, target)
+        loss.backward()
+        optimizer.step()
+
+
+def eval_model(model, valid_loader):
+    model.eval()
+    correct = 0
+    with torch.no_grad():
+        for batch_idx, (data, target) in enumerate(valid_loader):
+            if batch_idx * BATCH_SIZE >= N_VALID_SAMPLES:
+                break
+            data, target = data.view(data.size(0), -1).to(DEVICE), target.to(DEVICE)
+            output = model(data)
+            pred = output.argmax(dim=1, keepdim=True)
+            correct += pred.eq(target.view_as(pred)).sum().item()
+    accuracy = correct / min(len(valid_loader.dataset), N_VALID_SAMPLES)
+    return accuracy
+
+
+###################################################################################################
+# Define the objective function.
+def objective(trial):
+    train_loader = torch.utils.data.DataLoader(
+        torchvision.datasets.FashionMNIST(
+            DIR, train=True, download=True, transform=torchvision.transforms.ToTensor()
+        ),
+        batch_size=BATCH_SIZE,
+        shuffle=True,
+    )
+    valid_loader = torch.utils.data.DataLoader(
+        torchvision.datasets.FashionMNIST(
+            DIR, train=False, download=True, transform=torchvision.transforms.ToTensor()
+        ),
+        batch_size=BATCH_SIZE,
+        shuffle=True,
+    )
+
+    layers = []
+    in_features = 28 * 28
+    for i in range(3):
+        # Optimize the number of units of each layer and the initial learning rate.
+        out_features = trial.suggest_int("n_units_l{}".format(i), 4, 128)
+        layers.append(nn.Linear(in_features, out_features))
+        layers.append(nn.ReLU())
+        in_features = out_features
+    layers.append(nn.Linear(in_features, 10))
+    layers.append(nn.LogSoftmax(dim=1))
+    model = nn.Sequential(*layers).to(DEVICE)
+    # Sample the initial learning rate from [1e-5, 1e-1] in log space.
+    optimizer = torch.optim.Adam(
+        model.parameters(), trial.suggest_float("lr_init", 1e-5, 1e-1, log=True)
+    )
+
+    for step in range(10):
+        model.train()
+        train_model(model, optimizer, train_loader)
+
+        accuracy = eval_model(model, valid_loader)
+
+        # Report intermediate objective value.
+        trial.report(accuracy, step)
+
+        # Handle pruning based on the intermediate value.
+        if trial.should_prune():
+            raise optuna.TrialPruned()
+
+    return accuracy
+
+
+###################################################################################################
+# Run hyperparameter optimization with :class:`optuna.pruners.MedianPruner`.
+study = optuna.create_study(direction="maximize", pruner=optuna.pruners.MedianPruner())
+study.optimize(objective, n_trials=100, timeout=600)
+
+###################################################################################################
+# Visualization
```

Comment:
I feel `Visualization` after `Quick visualization` implies that this visualization is slow and fully-featured visualization. I think we'd like to introduce each plot function, so how about changing the section title?
The following is just an example:
```suggestion
# Plot functions
```